CODE-GUIDELINES.md

```diff
@@ -4,285 +4,68 @@ Please enforce these guidelines at all times. Small or large, call out what's in
+PatternFly follows [Harry Robert's CSS Guidelines](https://cssguidelin.es/) with some exceptions, deviations and additions.
 
-Everything else takes place inside the component or without impacting the previous two sections, and thus they come last.
+### Exceptions
 
-For a complete list of properties and their order, please see [Field CSS Manuals]( http://manuals.gravitydept.com/code/css/properties).
+PatternFly doesn't follow these rules:
 
-```sass
-.declaration-order {
-  // Positioning
-  position: absolute;
-  top: 0;
-  right: 0;
-  bottom: 0;
-  left: 0;
-  z-index: 100;
-
-  // Box-model
-  display: block;
-  float: right;
-  width: 100px;
-  height: 100px;
-
-  // Typography
-  font: normal 13px "Helvetica Neue", sans-serif;
-  line-height: 1.5;
-  color: #333;
-  text-align: center;
-
-  // Visual
-  background-color: #f5f5f5;
-  border: 1px solid #e5e5e5;
-  border-radius: 3px;
-
-  // Misc
-  opacity: 1;
-}
-```
+- [Table of contents](https://cssguidelin.es/#able-of-contents)
+- [Titling](https://cssguidelin.es/#titling)
+- [Anatomy of a Ruleset](https://cssguidelin.es/#anatomy-of-a-ruleset)
+- [Multi-line CSS](https://cssguidelin.es/#multi-line-css)
+- [Indenting](https://cssguidelin.es/#indenting)
+- [Meaningful Whitespace](https://cssguidelin.es/#meaningful-whitespace)
 
-<!-- ============================================================ -->
+### Deviations
 
-## Prefixed properties
+#### HTML
 
-PatternFly uses [Autoprefixer](https://github.com/postcss/autoprefixer) to deal with CSS vendor prefixes.
-
-**Do not add vendor prefixes**
-
-```sass
-// Bad
-.selector {
-  -webkit-box-shadow: 0 1px 2px $color;
-          box-shadow: 0 1px 2px $color;
-}
-
-// Good
-.selector {
-  box-shadow: 0 1px 2px $color;
-}
-```
-
-<!-- ============================================================ -->
-
-## Single declarations
-
-To improve error detection in instances where a rule set includes only one declaration, remove line breaks for readability and faster editing. Any rule set with multiple declarations should be split to separate lines.
-
-```sass
-// Single declarations on one line
-.span1 { width: 60px; }
-.span2 { width: 140px; }
-.span3 { width: 220px; }
-
-// Multiple declarations, one per line
-.sprite {
-  display: inline-block;
-  width: 16px;
-  height: 15px;
-  background-image: url(../img/sprite.png);
-}
+**Practicality over purity**. Strive to maintain HTML standards and semantics, but not at the expense of practicality. Use the least amount of markup with the fewest intricacies whenever possible.
 
-.icon           { background-position: 0 0; }
-.icon--home      { background-position: 0 -20px; }
-.icon--account   { background-position: 0 -40px; }
-```
+The CSS linter is PatternFly's single source of truth for CSS syntax, declaration order and other CSS rules like disallow type selectors, disallow vendor prefixes, and more.
```

Comment:
Should we write the rules in bullet point to make them more obvious?

Reply:
I'd say no. these are just examples not a list of the exact things we lint